FlowCrypt/Controllers/ImportKey/Enter Pass Phrase/EnterPassPhraseViewController.swift

```diff
@@ -178,23 +181,43 @@ extension EnterPassPhraseViewController {
         }
         showSpinner()
 
-        let matchingKeys = keyMethods.filterByPassPhraseMatch(keys: fetchedKeys, passPhrase: passPhrase)
+        let matchingKeys = keyMethods.filterByPassPhraseMatch(
+            keys: fetchedKeys,
+            passPhrase: passPhrase
+        )
 
-        guard matchingKeys.count > 0 else {
+        guard matchingKeys.isNotEmpty else {
             showAlert(message: "setup_wrong_pass_phrase_retry".localized)
             return
         }
 
-        let existedKeys = keysService.keys?.compactMap { $0.private } ?? []
-        let newKeys = fetchedKeys.map { $0.private }
-        let isKeyAlreadyAdded = Set(newKeys).isSubset(of: Set(existedKeys))
+        switch keyService.retrieveKeyDetails() {
+        case let .failure(error):
+            handleCommon(error: error)
+        case let .success(existedKeys):
+            checkIfKeyPossibleToAdd(with: existedKeys, and: passPhrase)
+        }
+    }
+
+    private func checkIfKeyPossibleToAdd(with existedKeys: [KeyDetails], and passPhrase: String) {
+        let isKeyAlreadyAdded = Set(fetchedKeys).isSubset(of: existedKeys)
```

Comment:
Sorry - while reading this source code, I got a different idea.

Instead of showing errors, when you encounter a private key with the same fingerprint, we would **replace** the stored key object instead of skipping it. So we would be doing an upsert, effectively.

So if the user imported 5 keys that were all the same, that would be fine - we'd process one by one and then finally the last one would end up stored in the database in the end.

Finally, we would show an alert.

1) if there were only new keys, no replacement: `Keys imported: <N>`
2) if there were only updates: `Keys imported: 0\nKeys updated: <N>`
3) if there was a mix of new keys and updates: `Keys imported: <N>\nKeys updated: <Y>`

We could show such alert regardless if this is first time setup or if adding keys from settings, either way it's good to know.

All of this will be achieved by comparing first fingerprint in each key, as you did here.

Thanks!